Tests/CSharp/src/Tests/AssetSystemTests.cs

```diff
@@ -1380,6 +1380,110 @@ public static void thirdPartyPackagedAssetIdentifierTest()
             Assert.AreEqual(assets[0].GetThirdPartyPlatformType(), Systems.EThirdPartyPlatform.UNITY);
 
         }
+#endif
+#if RUN_ALL_UNIT_TESTS || RUN_ASSETSYSTEM_TESTS || RUN_ASSETSYSTEM_COPY_ASSET_COLLECTION_TEST
+        [Test]
+        public static void copyAssetCollectionTest()
+        {
+            GetFoundationSystems(out var userSystem, out var spaceSystem, out var assetSystem, out _, out _, out _, out _, out _, out _, out _);
+
+            string sourceSpaceName = GenerateUniqueString("OLY-UNITTEST-SPACE-REWIND");
+            string destSpaceName = GenerateUniqueString("OLY-UNITTEST-SPACE-REWIND");
+            string testSpaceDescription = "OLY-UNITTEST-SPACEDESC-REWIND";
+            string testAssetCollectionName = GenerateUniqueString("OLY-UNITTEST-ASSETCOLLECTION-REWIND");
+            string testAssetName = GenerateUniqueString("OLY-UNITTEST-ASSET-REWIND");
+
+            var sourceAssetCollection = new Systems.AssetCollection();
+
+            // Log in
+            _ = UserSystemTests.LogIn(userSystem);
+
+            // Create 'source' space and asset collection
+            var sourceSpace = SpaceSystemTests.CreateSpace(spaceSystem, sourceSpaceName, testSpaceDescription, Systems.SpaceAttributes.Private, null, null, null);
+            {
+                CreateAssetCollection(assetSystem, sourceSpace, null, testAssetCollectionName, null, null, out sourceAssetCollection);
+                CreateAsset(assetSystem, sourceAssetCollection, testAssetName, null, null, out var sourceAsset);
+
+                var fileSource = new Systems.FileAssetDataSource
+                {
+                    FilePath = Path.GetFullPath("assets/test.json")
+                };
+                fileSource.SetMimeType("application/json");
+
+                assetSystem.UploadAssetDataOnProgress += (s, e) => LogDebug($"Uploading asset... ({e.Progress}%)");
+                using var result = assetSystem.UploadAssetData(sourceAssetCollection, sourceAsset, fileSource).Result;
+                var resCode = result.GetResultCode();
+
+                Assert.AreEqual(resCode, Systems.EResultCode.Success);
+            }
+
+            // Create 'dest' space and invoke the copy
+            var destSpace = SpaceSystemTests.CreateSpace(spaceSystem, destSpaceName, testSpaceDescription, Systems.SpaceAttributes.Private, null, null, null);
+            var destAssetCollections = new Common.Array<Systems.AssetCollection>();
```

Comment:
This needs to be disposed once you are done with it. It can initially be set to `null` as it looks like it's just being re-assigned later without the initial value being used anyway.

Reply:
Resolved!